File: python/mapbox_geocode.py
import __future__
import os, sys, json
try:
    # python 3
    from urllib.request import urlopen as urlopen
    from urllib.parse import quote_plus as quote_plus
except:
    # python 2
    from urllib import quote_plus as quote_plus
    from urllib2 import urlopen as urlopen

def geocode(mapbox_access_token, query):
    """
    Submit a geocoding query to Mapbox's geocoder.
    
    Args:
        mapbox_access_token (str): valid Mapbox access token with geocoding permissions
        query (str): input text to geocode
    """
    resp = urlopen('https://api.tiles.mapbox.com/v4/geocode/mapbox.places/{query}.json?access_token={token}'.format(query=quote_plus(query), token=mapbox_access_token))
    return json.loads(resp.read().decode('utf-8'))

if __name__ == '__main__':
    token = os.environ.get('MapboxAccessToken', False)
    if not token:
        print('environment variable MapboxAccessToken must be set')
        sys.exit(1)

    # geocode
    result = geocode(token, sys.argv[1])

    # print result
    print(json.dumps(result, indent=2))

File: node/mapbox-geocode.js
var https = require('https');

function geocode(mapboxAccessToken, query, callback) {
    https.get('https://api.tiles.mapbox.com/geocoding/v5/mapbox.places/' + encodeURIComponent(query) + '.json?access_token=' + mapboxAccessToken,
        function(response) {
            var body = '';
            response.on('data', function(d) {
                body += d;
            });
            response.on('error', function(e) {
                callback(e);
            });
            response.on('end', function() {
                callback(null, JSON.parse(body));
            });
        });
}

if (require.main === module) {
    if (!process.env.MapboxAccessToken) {
        console.log('environment variable MapboxAccessToken must be set');
        process.exit(1);
    }
    geocode(process.env.MapboxAccessToken, process.argv[2], function(err, result) {
        if (err) return console.log('Error: ' + err);
        console.log(JSON.stringify(result, null, 2));
    });
}

module.exports = geocode;